src/hotspot/share/c1/c1_GraphBuilder.cpp

```diff
@@ -4442,6 +4438,20 @@ void GraphBuilder::append_char_access(ciMethod* callee, bool is_store) {
 }
 
 void GraphBuilder::append_alloc_array_copy(ciMethod* callee) {
+  {
+    // Peek at receiver
+    Value recv = apop();
```

Comment:
What about this instead and remove the `apush(recv)`?

```
Value recv = state()->stack_at(state()->stack_size() - callee->arg_size());
```

Reply:
Yes, that's a better idiom for receiver because it takes possible arguments into account.